spacy/language.py

```diff
@@ -278,49 +411,57 @@ def get_pipe(self, name):
                 return component
         raise KeyError(Errors.E001.format(name=name, opts=self.pipe_names))
 
-    def create_pipe(self, name, config=dict()):
-        """Create a pipeline component from a factory.
-
-        name (str): Factory name to look up in `Language.factories`.
-        config (dict): Configuration parameters to initialise component.
-        RETURNS (callable): Pipeline component.
-
-        DOCS: https://spacy.io/api/language#create_pipe
-        """
-        if name not in self.factories:
-            raise KeyError(Errors.E002.format(name=name))
-        factory = self.factories[name]
-
-        # transform the model's config to an actual Model
-        factory_cfg = dict(config)
-
-        # check whether we have a proper model config, ignore if the type is wrong
-        if "model" in factory_cfg and not isinstance(factory_cfg["model"], dict):
-            warnings.warn(
-                Warnings.W099.format(type=type(factory_cfg["model"]), pipe=name)
-            )
-
-        # refer to the model configuration in the cfg settings for this component
-        elif "model" in factory_cfg:
-            self.config[name] = {"model": factory_cfg["model"]}
-
-        # create all objects in the config
-        factory_cfg = registry.make_from_config({"config": factory_cfg}, validate=True)[
-            "config"
-        ]
-        model = factory_cfg.get("model", None)
-        if model is not None:
-            del factory_cfg["model"]
-        return factory(self, model, **factory_cfg)
+    def create_pipe(
+        self,
+        factory_name: str,
+        name: Optional[str] = None,
+        config: Optional[Dict[str, Any]] = SimpleFrozenDict(),
+        validate: bool = True,
+    ) -> Callable[[Doc], Doc]:
+        name = name or factory_name
+        if not isinstance(config, dict):
+            err = Errors.E962.format(style="config", name=name, cfg_type=type(config))
+            raise ValueError(err)
+        if not srsly.is_json_serializable(config):
+            raise ValueError(Errors.E961.format(config=config))
+        if not self.has_factory(factory_name):
+            raise ValueError(Errors.E002.format(name=factory_name))
+        pipe_meta = self.get_factory_meta(factory_name)
+        config = config or {}
+        # This is unideal, but the alternative would mean you always need to
+        # specify the full config settings, which is not really viable.
+        if pipe_meta.default_config:
+            config = util.deep_merge_configs(config, pipe_meta.default_config)
```

Comment:
Should we add a check here, and warn if default values have been merged/filled in?

Reply:
Yes, I think this part could definitely benefit from more warnings 👍 We should also add `W097` and `W098` back here. 

We could probably just add a simple check that compares the default config to the regular config and only attempt to deep merge if they're different at all.

Not sure what the most efficient way is to check whether _some_ values have been added... like, we only care about block that don't reference registered functions or blocks that reference the same registered functions. Maybe the `deep_merge_configs` helper could just keep a count? Or maybe I'm overthinking this 😅